bootloaders/uart/uart_boot.c

```diff
@@ -0,0 +1,152 @@
+#include <stdio.h>
+#include <stdlib.h>
+#include "pico/stdlib.h"
+#include "hardware/uart.h"
+#include "pico/bootrom.h"
+#include "boot/picobin.h"
+#include "hardware/flash.h"
+
+// UART defines for uart boot
+#define UART_ID uart1
+#define BAUD_RATE 1000000
+
+// Use pins 4 and 5 for uart boot
+#define UART_TX_PIN 4
+#define UART_RX_PIN 5
+
+// Use pin 3 for the RUN pin on the other chip
+#define RUN_PIN 3
+
+
+void reset_chip() {
+    // Toggle run pin
+    gpio_put(RUN_PIN, false);
+    sleep_ms(1);
+    gpio_put(RUN_PIN, true);
+}
+
+
+void uart_boot() {
+    uint knocks = 0;
+    while (true) {
+        // Send the knock sequence
+        uart_putc_raw(UART_ID, 0x56);
+        uart_putc_raw(UART_ID, 0xff);
+        uart_putc_raw(UART_ID, 0x8b);
+        uart_putc_raw(UART_ID, 0xe4);
+        uart_putc_raw(UART_ID, 'n');
+
+        if (uart_is_readable_within_us(UART_ID, 1000)) {
+            char in = uart_getc(UART_ID);
+            printf("%c\n", in);
```

Comment:
Should this check that `in == 'n'` ?

Reply:
Yep - I've added checks that will call `reset_chip()` if it's the incorrect character for all these read backs